data/recommendation/notify.js

```diff
@@ -139,17 +139,33 @@ class Notify {
           <p>
             You dodged the bullet with not having an account here.
           </p>
-          <p>
-            Protect yourself with Firefox Sync for your passwords
-            and you will be informed each time one of your accounts might
-            be at risk.
-          </p>
+          ${this.createSyncOffer()}
         </div>
         ${this.createSignupFooter()}
       </div>
     `;
   }
 
+  createSyncOffer() {
```

Comment:
One thing that actually makes it difficult to track what is being output is that you keep splitting your output functions into several smaller functions that are only called once. Every time your output functions defer to another function, I have to store in my head what that function outputs and figure out what the final HTML is.

Normally factoring out into a function has several benefits to reducing cognitive load:
- Reuse of code that would otherwise be repeated.
- Encapsulation of a well-understood operation to reduce the visual noise in places where the function is used. Think like `isSyncEnabled`: it clearly returns a boolean value indicating if sync is enabled, and no knowledge of _how_ it determined that is necessary.

But in this case, the first point doesn't apply since you're not reusing it, and the second point doesn't apply because generating the HTML for the offer isn't a well-understood operation. It'd be better to just keep this logic directly in the parent method:

```js
return yo`
  <p>
    Foo bar baz
    ${this.specialOffer && 'Special!'}
  </p>
`
```

Another place you'll see this logic a lot is in server-side template languages; I've seen attempts at being clever with splitting off things into lots of smaller macros, and it almost always ends up being harder to understand.